python_modules/dagster/dagster/_core/definitions/auto_materialize_rule.py

```diff
@@ -676,6 +691,42 @@ def evaluate_for_asset(self, context: RuleEvaluationContext) -> RuleEvaluationRe
         return []
 
 
+@whitelist_for_serdes
+class SkipOnBackfillInProgressRule(
+    AutoMaterializeRule,
+    NamedTuple(
+        "_SkipOnBackfillInProgressRule", [("skip_all_partitions_of_backfilling_asset", bool)]
+    ),
+):
+    @property
+    def decision_type(self) -> AutoMaterializeDecisionType:
+        return AutoMaterializeDecisionType.SKIP
+
+    @property
+    def description(self) -> str:
```

Comment:
this description should change based on the value of the boolean. I'm thinking

"targeted by an in-progress backfill"

and

"part of an asset targeted by an in-progress backfill"

The second one is a bit weirder, but we're trying to make these fit into the framework of "This asset will be skipped if it is _____"

Reply:
`targeted by an in-progress backfill` sounds good to me

Reply:
ah in particular, I think we should return different strings depending on the value of self.all_partitions, similar to what we do with the `SkipOnNotAllParentsUpdatedRule`

Reply:
ah yes, of course -- apologies for misreading. this has been updated